lua/lspconfig/volar.lua

```diff
@@ -0,0 +1,103 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local function get_typescript_server_path(root_dir)
+  local project_root = util.find_node_modules_ancestor(root_dir);
+  return project_root and (project_root .. '/node_modules/typescript/lib/tsserverlibrary.js') or '';
+end
+
+local server_name = 'volar'
+local bin_name = 'volar-server'
+
+-- https://github.com/johnsoncodehk/volar/blob/master/packages/shared/src/types.ts
+local volar_init_options = {
+  typescript = {
+    serverPath = get_typescript_server_path(vim.fn.getcwd()),
+  },
+  languageFeatures={
+    references = true,
+    definition = true,
+    typeDefinition = true,
+    callHierarchy = true,
+    hover = true,
+    rename = true,
+    renameFileRefactoring = true,
+    signatureHelp = true,
+    codeAction = true,
+    completion={
+      defaultTagNameCase='both',
+      defaultAttrNameCase='kebabCase',
+    },
+    schemaRequestService  = true ,
+    documentHighlight = true,
+    documentLink = true ,
+    codeLens = true,
+    semanticTokens = true,
+    diagnostics = true,
+  },
+  documentFeatures={
+    selectionRange = true,
+    foldingRange = true,
+    linkedEditingRange = true,
+    documentSymbol = true,
+    documentColor = true,
+    documentFormatting = {
+      defaultPrintWidth = 100,
+    },
+  }
+}
+
+-- https://github.com/johnsoncodehk/volar/blob/master/package.json#L56
+local volar_settings = {
+  volar= {
+    codeLens= {
+      references = true,
+      pugTools = true,
+      scriptSetupTools = true,
+    },
+    lowPowerMode = false,
+    tsPluginStatus = false,
+    checkVueTscVersion = false,
+    preview= {
+      port = 3333,
+      backgroundColor = "fff",
+      transparentGrid = false,
+    }
+  }
+}
+
+configs[server_name] = {
+  default_config = {
+    cmd = { bin_name, '--stdio' },
+    filetypes = { 'vue' },
+    root_dir = util.root_pattern('package.json'),
+    init_options = volar_init_options,
+    settings = volar_settings,
+  },
+  docs = {
+    package_json = 'https://raw.githubusercontent.com/johnsoncodehk/volar/master/packages/server/package.json',
+    description = [[
+    https://github.com/johnsoncodehk/volar/tree/master/packages/server
+    Volar language server for Vue
+
+    Volar can be installed via npm
+    ```sh
+    npm install -g @volar/server
+    vue-server --stdio
```

Comment:
Do you need to run vue-server --stdio during installation for some reason?  That's how this reads.

Reply:
done